Parsers/ASimFileEvent/Parsers/vimFileEventMicrosoftWindowsEvents.yaml

```diff
@@ -34,66 +70,179 @@ ParserQuery: |
+    union isfuzzy=false (
+    WindowsEvent
+        | where EventID == 4670
+            and EventData.ObjectType == "File"
+        | project TimeGenerated
+            , Computer
+            , EventID
+            , SubjectUserSid = tostring(EventData.SubjectUserSid)
+            , AccountType = tostring(EventData.AccountType)
+            , SubjectUserName = tostring(EventData.SubjectUserName)
+            , SubjectAccount = tostring(EventData.SubjectAccount)
+            , SubjectLogonId = tostring(EventData.SubjectLogonId)
+            , SubjectDomainName = tostring(EventData.SubjectDomainName)
+            , ObjectName = tostring(EventData.ObjectName)
+            , ProcessName = tostring(EventData.ProcessName)
+            , ProcessId = tostring(EventData.ProcessId)
+            , HandleId = tostring(EventData.HandleId)
+        | extend SubjectAccount = strcat(SubjectDomainName,"\\",SubjectUserName)
+        | extend AccountType = "Unknown"
+        | project-away SubjectDomainName
+    )
+  , (
+    SecurityEvent
+        | where EventID == 4670
+            and ObjectType == "File"
+        | project TimeGenerated
+                , EventID
+                , SubjectUserSid
+                , AccountType
+                , Computer
+                , SubjectUserName
+                , SubjectAccount
+                , SubjectLogonId
+                , ObjectName
+                , ProcessId
+                , ProcessName
+                , HandleId
+    )
+        | extend ActingProcessName = ProcessName
+            , ActingProcessId = tostring(toint(ProcessId))
+            , TargetFilePath = ObjectName
+        | project-away ProcessId
+        | project-rename Process = ProcessName
+            , FilePath = ObjectName
+            , FileSessionId = HandleId
+    ;
+  union isfuzzy=false
+    4663events
+    , 5140events
+    , 4670events
+        | lookup ActorUserTypeLookup on AccountType
+        | lookup SIDCommonLookup on $left.SubjectUserSid == $right.sid
+        | extend ActorUsername = iff (SubjectUserName == "-", username, SubjectAccount)
+            , ActorUsernameType = iff(SubjectUserName == '-',type, 'Windows')
+            , EventStartTime = TimeGenerated
+            , EventEndTime = TimeGenerated
+            , TargetFilePathFormat = "Windows Local"
+            , EventOriginalType = tostring(EventID)
+        | project-away EventID, AccountType, type, username
+        | project-rename ActorUserId = SubjectUserSid
+            , DvcHostname = Computer
+            , ActorSessionId = SubjectLogonId
+        | extend EventSchema = "FileEvent"
+            , EventSchemaVersion = "0.1.1"
+            , EventResult = "Success"
+            , EventCount = int(1)
+            , EventVendor = 'Microsoft'
+            , EventProduct = 'Security Events'
+            , Dvc = DvcHostname
+            , ActorWindowsUsername = ActorUsername
+            , User = ActorUsername
+            , ActorUserSid = ActorUserId
```

Comment:
Please have a look at the data test results:
![image](https://github.com/Azure/Azure-Sentinel/assets/97222872/f470b13b-9240-4055-bc5e-bdc54f00d10d)